### DISCUSSION TITLE
/usr/local/opt/rabbitmq/sbin/rabbitmq-server: line 155:  2285 Segmentation fault: 11  start_rabbitmq_server "$@"

### DISCUSSION THREAD

**Discussion by kavyabhargava** [rabbitmq/rabbitmq-server#5115] [Other]
Getting segmentation error again and again when trying to run rabbitmq. I have a M1pro MacBook

I did brew install rabbit mq. I got no errors but then I ran 
`CONF_ENV_FILE="/usr/local/etc/rabbitmq/rabbitmq-env.conf" /usr/local/opt/rabbitmq/sbin/rabbitmq-server`

and I got this error. It was running perfectly till yesterday but suddenly it is showing this segmentation error. I have tried reinstalling it, uninstalling it and everything

**Comment by lukebakken** [comment]
This is something to do with the Erlang build you are using. How did you install Erlang and RabbitMQ? Did you compile Erlang from source?

**Comment by kavyabhargava** [comment]
@lukebakken I did not install Erlang(It is already installed.["erlang 25.0.2 is already installed"]. I installed rabbitmq through brew install rabbitmq

"Did you compile Erlang from source?" I'm sorry I am not sure what this means. I ran rabbitmq on the terminal globally rather in project directory.Also tried in project directory got same error

**Reply by lukebakken** [reply]
This is probably an issue with your specific OS X and M1 system. Segmentation faults come from the Erlang VM itself.

> It was running perfectly till yesterday

What changed? Did you install operating system updates?

We need the following information:
* Complete RabbitMQ log file
* **All** of your configuration files
* System report - https://support.apple.com/guide/system-information/get-system-information-syspr35536/mac

**Reply by kavyabhargava** [reply]
Could you tell me command for first two points?

**Comment by michaelklishin** [comment]
I run RabbitMQ, both binary builds and nodes built from source, on Erlang 23 through 25 on an M1 Mac almost every single day.

You need to delete your Erlang installation and reinstall. I don't use Homebrew to provision Erlang
but [their Erlang formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/erlang.rb) does list ARM64-specific bottles (binary builds).

What segfaults is the Erlang runtime, RabbitMQ does not allocate or access memory directly.

**Reply by kavyabhargava** [reply]
I just did that and ran rabbitmq again . Getting same error

**Reply by lukebakken** [reply]
> I just did that 

It's _important_ that you tell us HOW you do things, not just that you did them, and provide the output of the commands you run.

At this point I suggest compiling Erlang yourself. This is a popular tool to do so:

https://github.com/kerl/kerl

**Reply by kavyabhargava** [reply]
So first I did the brew uninstall Erlang but it was not working giving : Error: 
```
Refusing to uninstall /usr/local/Cellar/erlang/25.0.2
because it is required by rabbitmq, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies erlang

```
but then I did the `brew uninstall --ignore-dependencies erlang`. It uninstalled erlang. Then :
I did `brew install erlang` as specified in their website https://www.erlang.org/downloads

output summary:
```
 Summary
  /usr/local/Cellar/erlang/25.0.2: 7,866 files, 486.6MB
```

**Reply by michaelklishin** [reply]
I have never seen a `kerl`-built Erlang 25 segfault on an M1 Mac (if built on the same Mac, obivously). We cannot speak for how the Erlang bottles for ARM64 are produced.

You can try both to compare using the [binary builds of RabbitMQ](https://rabbitmq.com/install-generic-unix.html). They only need a compatible Erlang version in `PATH`.

There isn't much else we can suggest, RabbitMQ does not allocate memory directly so our own code cannot be directly resposible for a segfault.

**Comment by mobweb** [comment]
@kavyabhargava: Did you ever find a solution for this? Exact same issue and specification here.
